scrapper.py

```diff
@@ -26,38 +36,133 @@ class Crawler:
+    else:
+        os.makedirs(base_path)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+    seed_urls = config['seed_urls']
+    max_articles = config['total_articles_to_find_and_parse']
+    for seed_url in seed_urls:
+        if not re.match(r'https?://', seed_url):
+            raise IncorrectURLError
+    if not seed_urls:
+        raise IncorrectURLError
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if not isinstance(max_articles, int):
+        raise IncorrectNumberOfArticlesError
+    if max_articles > 100:
+        raise NumberOfArticlesOutOfRangeError
+    if max_articles == 0 or max_articles < 0:
+        raise IncorrectNumberOfArticlesError
+    prepare_environment(ASSETS_PATH)
+    return seed_urls, max_articles
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def parse(self):
+        self.article = Article(self.article_url, self.article_id)
+        headers = {'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) '
+                                 'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36'}
+        response = requests.get(self.article_url, headers)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        table_rows = article_bs.find('iframe', class_='pdf')
+        print(table_rows['data-src'])
+        url_of_pdf = table_rows['data-src']
+        pdf_raw_file = PDFRawFile(url_of_pdf, self.article_id)
+        pdf_raw_file.download()
+        self.article.text = pdf_raw_file.get_text()
+        print(len(self.article.text))
+        if 'Литература' in self.article.text:
+            new_list = self.article.text.split('Литература')
+            new_list_without_literature = new_list[:-1]
+            new_list_without_literature = ''.join(new_list_without_literature)
+            self.article.text = new_list_without_literature
+        print(len(self.article.text))
+
+    def _fill_article_with_meta_information(self, article_bs):
+        author_bs = article_bs.find('span', class_='field__item-wrapper')
+        if not author_bs:
+            author_bs = 'NOT FOUND'
+        else:
+            author_bs = author_bs.get_text()
+        title_of_the_article_bs = article_bs.find('title')
+        title_of_the_article_bs = title_of_the_article_bs.get_text()
+        data_bs = article_bs.find('div', class_='node__content clearfix')
+        data = data_bs.find('div',
```

Comment:
is there any more reliable way to get text?

Reply:
Could you explain please what exactly should I do here?

Reply:
the markers that you are using for finding text are very fragile. imagine, I am an author of the journal and decided to make margins for my text larger, for example 20px. Your code will never find text then. Do you see what I am talking about?